DTS-STN/canadian-dental-care-plan
# Updating application year entity, repository, DTO and mapper based on updated specs

### Description
Based on Interop v1.0.12 specs that adds endpoint to retrieve application year config.

DTO has been updated to include fields that the frontend application will eventually need.

### Related Azure Boards Work Items
AB#4860

### Screenshots (if applicable)
![image](https://github.com/user-attachments/assets/9747044f-eb5c-4cc4-ae79-f1993ba8bc01)

### Checklist

### Test Instructions
Run frontend locally and go through the first step of the `/apply` flow (before terms and conditions). Verify that the application year id and tax year are saved in state.
